src/Workspaces/Core/Portable/Execution/AssetBuilder.cs

```diff
@@ -75,6 +76,13 @@ public Asset Build(TextDocumentState state, SourceText text, CancellationToken c
             return asset;
         }
 
+        public Asset Build(OptionSet options, string language, CancellationToken cancellationToken)
+        {
+            // get around issue where this can't be captured in struct
+            var local = this;
```

Comment:
if this is just causing a heavy hoisted allocation, then why not just use a class in the first place?